homeassistant/components/husqvarna_automower/switch.py

```diff
@@ -92,3 +114,100 @@ async def async_turn_on(self, **kwargs: Any) -> None:
+    def stay_out_zones(self) -> StayOutZones:
+        """Return all stay out zones."""
+        if TYPE_CHECKING:
+            assert self.mower_attributes.stay_out_zones is not None
+        return self.mower_attributes.stay_out_zones
+
+    @property
+    def stay_out_zone(self) -> Zone:
+        """Return the specific stay out zone."""
+        return self.stay_out_zones.zones[self.stay_out_zone_uid]
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return self.stay_out_zone.enabled
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available and the zones are not `dirty`."""
+        return super().available and not self.stay_out_zones.dirty
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the switch off."""
+        try:
+            await self.coordinator.api.commands.switch_stay_out_zone(
+                self.mower_id, self.stay_out_zone_uid, False
+            )
+            # As there are no updates from the websocket regarding stay out zone changes,
+            # we need to wait 5s and then poll the API.
+            await asyncio.sleep(5)
+            await self.coordinator.async_request_refresh()
+        except ApiException as exception:
+            raise HomeAssistantError(
+                f"Command couldn't be sent to the command queue: {exception}"
+            ) from exception
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the switch on."""
+        try:
+            await self.coordinator.api.commands.switch_stay_out_zone(
+                self.mower_id, self.stay_out_zone_uid, True
+            )
+            # As there are no updates from the websocket regarding stay out zone changes,
+            # we need to wait 5s and then poll the API.
+            await asyncio.sleep(5)
+            await self.coordinator.async_request_refresh()
+        except ApiException as exception:
+            raise HomeAssistantError(
+                f"Command couldn't be sent to the command queue: {exception}"
+            ) from exception
+
+
+@callback
+def async_remove_entities(
+    hass: HomeAssistant,
+    coordinator: AutomowerDataUpdateCoordinator,
+    config_entry: ConfigEntry,
+    mower_id: str,
+) -> None:
+    """Remove deleted stay-out-zones from Home Assistant."""
+    entity_reg = er.async_get(hass)
+    active_zones = set()
+    _zones = coordinator.data[mower_id].stay_out_zones
+    if _zones is not None:
+        for zones_uid in _zones.zones:
+            uid = f"{mower_id}_{zones_uid}_stay_out_zones"
+            active_zones.add(uid)
+        for entity_entry in er.async_entries_for_config_entry(
+            entity_reg, config_entry.entry_id
+        ):
+            if entity_entry.unique_id.split("_")[0] == mower_id:
+                if entity_entry.unique_id.endswith("stay_out_zones"):
+                    if entity_entry.unique_id not in active_zones:
```

Comment:
This can be combined into one if statement by joining all statements with `and`.
Also it's strange that you use one time split and the other time endsWith. You could also use the the last value of the split instead